tests/main/interfaces-pulseaudio/task.yaml

```diff
@@ -1,98 +1,59 @@
 summary: Ensure that the pulseaudio interface works
 
-# Classic Ubuntu is sufficient to test the feature
-systems: [ubuntu-1*-*64, ubuntu-2*-*64]
+# Classic Ubuntu is sufficient to test the feature,
+# except Ubuntu 14.04 where systemd is not supported.
+systems: [ubuntu-16.04-*, ubuntu-18.04-*, ubuntu-19.10-*, ubuntu-20-*]
 
 kill-timeout: 3m
 
 environment:
     PLAY_FILE: "/snap/test-snapd-pulseaudio/current/usr/share/sounds/alsa/Noise.wav"
-    PA_TEST_LOG: /home/test/pulseaudio.log
 
 prepare: |
-    # FIXME: This test is broken and should be ported to session-tool, systemd
-    # --user starts pulseaudio for us so we are always racing with it (whoever
-    # acquires the socket file wins).
-    #
-    # To prevent this from breaking this test mask pulseaudio.{socket,service}
-    # in all user sessions.
-    systemctl --user --global mask pulseaudio.{socket,service}
-
-    #shellcheck source=tests/lib/pkgdb.sh
-    . "$TESTSLIB"/pkgdb.sh
-    snap install --edge test-snapd-pulseaudio
+    truncate --size=0 defer.sh
+    chmod +x defer.sh
 
+    # Install pulseaudio.
     apt-get update
     apt-get install -y pulseaudio pulseaudio-utils
+    echo "apt-get autoremove --purge -y pulseaudio pulseaudio-utils" >>defer.sh
 
-    echo "Create XDG_RUNTIME_DIR=/run/user/12345"
-    # shellcheck disable=SC2174
-    mkdir -m 700 -p /run/user/12345 || true
-    chown test:test /run/user/12345
-
-    # ensure we have a clean pulse directory
-    test -d /home/test/.config && mv /home/test/.config /home/test/.config.spread
-    mkdir -m 700 /home/test/.config
-    mkdir -m 700 /home/test/.config/pulse
-    chown test:test /home/test/.config /home/test/.config/pulse
-
-    # prepare the script for pulseaudio daemon
-    cat <<'EOF' > /home/test/pulse-test.pa
-    .fail
-    load-module module-null-sink sink_name=void
-    set-default-sink void
```

Comment:
The test config set up a default null sink. Now we're back to the distro config which may use a real sound device when the test is run on real hardware by @sergiocazzolato

Reply:
I think that's fine if it works. This is better that it actually tests the distribution setting. Before we artificially injected the policy module so our test would pass if the actual Ubuntu config was buggy and would result in a security violation.

Reply:
I would rather us not use the distro config _specifically because_ there may be bugs there. In general where it's not a lot of work (and in this case we have already done this work) replacing distro things that can become broken randomly at some point and thus are unstable sounds like a good thing for us, as we're not trying to test the distro we just want to test our things that interact with the distro. Making the distro less of a moving target is a good thing in my mind.

Reply:
I guess it depends on what we want to test, our own side or the integration of the distribution with our own side. Perhaps we just need to have a pair of tests (one with canned config, one with stock config).

Reply:
Btw, when the distribution drops the confinement patches *I* will say told you so :) This will catch a real regression :D

Reply:
ah you know I had honestly forgotten that the config we are specifying loads the snap policy module, I thought that was compiled into pulseaudio via patches and thus couldn't be turned off via config, but it appears that indeed you are right the mediation could be turned off via distro config... In this case you're right and I think we should be using distro config for this test.